src/transforms/GetExternalFunctions.cpp

```diff
@@ -0,0 +1,117 @@
+/**
+ * LLVM transformation pass to resolve indirect calls
+ *
+ * The transformation performs "devirtualization" which consists of
+ * looking for indirect function calls and transforming them into a
+ * switch statement that selects one of several direct function calls
+ * to execute. Devirtualization happens if a pointer analysis can
+ * resolve the indirect calls and compute all possible callees.
+ **/
+
+#include <iostream>
+#include <fstream>
+
+#include "llvm/Pass.h"
+//#include "llvm/Analysis/CallGraph.h"
+#include "llvm/IR/MDBuilder.h"
+#include "llvm/Support/CommandLine.h"
+#include "llvm/Support/raw_ostream.h"
+
+#include "seadsa/CompleteCallGraph.hh"
+#include "seadsa/InitializePasses.hh"
+
+
+namespace previrt {
+    namespace transforms {
+
+        using namespace llvm;
+
+        /** 
+         ** Resolve indirect calls by one direct call for possible callee
+         ** function 
+         **/
+        class GetExternalFunctions : public ModulePass {
+            public:
+                static char ID;
+
+                std::string getModuleName(std::string path){
```

Comment:
You should try not use std::string. LLVM provides StringRef which are much cheaper